### PR TITLE
fix(sdl): use service name to find expose endpoints

### LINKED ISSUE #1215: sdl: compute profile name assumed to be service name
`GetDeploymentGroups()` [uses the profile name](https://github.com/ovrclk/akash/blob/9b11cf65d6ec58ef31a66b639fc20f9d825b795d/sdl/v2.go#L123) when determining endpoints to include in a group instead of the service name.

This causes sdl files like the one below to not have any endpoints in their groups or orders.  Wen sending the manifest, validation fails because of an endpoint count mismatch ([`Manifest()`](https://github.com/ovrclk/akash/blob/9b11cf65d6ec58ef31a66b639fc20f9d825b795d/sdl/v2.go#L170) apparently does the right thing).

```yaml
---
version: "2.0"
services:
  web:
    image: ourwebapp
    expose:
      - port: 8080
        as: 80
        to:
          - global: true
profiles:
  compute:
    frontend:
      resources:
        cpu:
          units: 1
        memory:
          size: 2Gi
        storage:
          size: 32Gi
  placement:
    westcoast:
      pricing:
        frontend:
          denom: uakt
          amount: 100

deployment:
  web:
    westcoast:
      profile: frontend
      count: 1
```

### PR DESCRIPTION
**Pull request by troian**
fixes #1215

Signed-off-by: Artur Troian <troian.ap@gmail.com>

### CODE REVIEW

**Review by boz — COMMENTED** [review]
mind adding a test for this?

**Comment by troian** [comment]
test added

File: sdl/_testdata/deployment-svc-mismatch.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+version: "2.0"
```
Comment (hydrogen18) [inline sdl/_testdata/deployment-svc-mismatch.yaml:2]: What is the error in this file that leads to the test case? I don't see it immediately.

File: sdl/v2_test.go [hunk 2]
```diff
@@ -192,3 +193,10 @@ func Test_v1_Parse_ProfileNameNotServiceName(t *testing.T) {
 	require.NoError(t, err)
 	assert.Len(t, mani.GetGroups(), 1)
 }
+
+func Test_v2_Parse_DeploymentNameServiceNameMismatch(t *testing.T) {
+	sdl, err := ReadFile("./_testdata/deployment-svc-mismatch.yaml")
+	require.Nil(t, sdl)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "no service profile named")
```
Comment (boz) [inline sdl/v2_test.go:201]: The problem was that it was not registering endpoints, everything else worked fine.  It was looking for endpoints by compute profile name, not service name.

```yaml
services:
  web:
    # ...
profiles:
  compute:
    bew:
      # ...
```

**Review by boz — CHANGES_REQUESTED** [review]

**Review by hydrogen18 — APPROVED** [review]

File: sdl/v2_test.go (hunk 2 above)
Comment (troian) [reply sdl/v2_test.go:201]: i probably missed point here. fix makes endpoints iteration via service name. if deployment points to non-existing service it fails on `ReadFile`

valid
```yaml
services:
  web:
    # ...
profiles:
  compute:
    bew:
      # ...
  placement:
    westcoast:
      # ...
deployment:
  web:
    westcoast:
      profile: bew
```

invalid
```yaml
services:
  web:
    # ...
profiles:
  compute:
    bew:
      # ...
  placement:
    westcoast:
      # ...
deployment:
  bew:
    westcoast:
      profile: bew
```

Comment (boz) [reply sdl/v2_test.go:201]: It wasn't failing it just wasn't picking up the endpoints.

Comment (boz) [reply sdl/v2_test.go:201]: and yeah, your first example would have triggered it.

Comment (troian) [reply sdl/v2_test.go:201]: updated test

**Review by boz — APPROVED** [review]
:ship:

**Pull request merged**